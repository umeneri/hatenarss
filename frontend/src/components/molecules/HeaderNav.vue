<template>
  <header>
    <nav class="navbar has-shadow" role="navigation" aria-label="main navigation">
      <div class="navbar-brand">
        <div class="navbar-item">
          <img src="/assets/logo.png"/>
        </div>
        <a role="button"
           class="navbar-burger burger"
           aria-label="menu"
           aria-expanded="false"
           data-target="navbarManu"
           @click="toggleMenu"
           :class="{'is-active': isMenuActive}">
          <span aria-hidden="true"></span>
          <span aria-hidden="true"></span>
          <span aria-hidden="true"></span>
        </a>
      </div>

      <div id="navbarManu" class="navbar-menu is-dark" :class="{'is-active': isMenuActive}">
        <div class="navbar-start">
          <RouterLink
            v-for="menu in menus"
            :key="menu.name"
            :to="menu.path"
            class="navbar-item">
              <span class="icon"><i :class="iconClasses(menu)"></i></span>
              <span>{{ menu.name }}</span>
          </RouterLink>
        </div>
      </div>
    </nav>
  </header>
</template>

<script>
export default {
  props: {
    menus: {
      type: Array
    }
  },
  data () {
    return {
      isMenuActive: false
    }
  },
  methods: {
    toggleMenu () {
      this.isMenuActive = !this.isMenuActive
    },
    closeMenu () {
      this.isMenuActive = false
    },
    selectMenu (selectedMenu) {
      this.menus.forEach(menu => {
        menu.isActive = (menu.name === selectedMenu.name);
      });
    },
    iconClasses (menu) {
      return `fa fa-${menu.icon}`
    }
  }
}
</script>

<style lang="sass" scoped>
$navbar-background-color: #363636
$navbar-item-color: hsl(0, 0%, 100%)
$navbar-item-hover-color: hsl(0, 0%, 29%)
$navbar-item-hover-background-color: hsl(0, 0%, 98%)

@import '~bulma'
</style>

