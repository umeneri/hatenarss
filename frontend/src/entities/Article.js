export default class Article {
  constructor (item) {
    this.bookmarkcount = item.bookmarkcount
    this.datetime = item.datetime
    this.description = item.description
    this.imageurl = item.imageurl
    this.link = item.link
    this.title =  item.title
    this.hasVisibleDescription = false
  }

  toggleDescription () {
    this.hasVisibleDescription = !this.hasVisibleDescription
  }
}
