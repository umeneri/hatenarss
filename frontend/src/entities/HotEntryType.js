export const HOT_ENTRY_TYPE = {
  HOT_ENTRY: {
    name: '総合',
    category: 'hotentry'
  },
  SOCIAL: {
    name: '社会',
    category: 'social'
  },
  ECONOMICS: {
    name: '政治と経済',
    category: 'economics'
  },
  LIFE: {
    name: '暮らし',
    category: 'life'
  },
  KNOWLEDGE: {
    name: '学び',
    category: 'knowledge'
  },
  IT: {
    name: 'テクノロジー',
    category: 'it'
  },
  ENTERTAINMENT: {
    name: 'エンタメ',
    category: 'entertainment'
  },
  GAME: {
    name: 'アニメとゲーム',
    category: 'game'
  },
  FUN: {
    name: 'おもしろ',
    category: 'fun'
  },
  VIDEO: {
    name: '動画',
    category: 'video'
  }
}
