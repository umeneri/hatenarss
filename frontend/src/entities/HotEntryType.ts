export const HOT_ENTRY_TYPE:{ [index:string] : {name: string, path: string} } = {
  HOT_ENTRY: {
    name: '総合',
    path: 'hotentry'
  },
  SOCIAL: {
    name: '社会',
    path: 'social'
  },
  ECONOMICS: {
    name: '政治と経済',
    path: 'economics'
  },
  LIFE: {
    name: '暮らし',
    path: 'life'
  },
  KNOWLEDGE: {
    name: '学び',
    path: 'knowledge'
  },
  IT: {
    name: 'テクノロジー',
    path: 'it'
  },
  ENTERTAINMENT: {
    name: 'エンタメ',
    path: 'entertainment'
  },
  GAME: {
    name: 'アニメとゲーム',
    path: 'game'
  },
  FUN: {
    name: 'おもしろ',
    path: 'fun'
  },
  VIDEO: {
    name: '動画',
    path: 'video'
  }
}
