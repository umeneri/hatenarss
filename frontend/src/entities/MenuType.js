export const MENU_TYPE = {
  RANKING: {
    name: 'Ranking',
    path: '/ranking/daily',
    icon: 'home',
    isActive: true,
  },
  HOT_ENTRY: {
    name: 'HotEntry',
    path: '/hotentry/hotentry',
    icon: 'fire',
    isActive: false,
  },
  ABOUT: {
    name: 'About',
    path: '/about',
    icon: 'info',
    isActive: false,
  }
}
