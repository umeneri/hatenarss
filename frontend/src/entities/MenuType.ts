import MenuItem from '@/interfaces/MenuItem'

export const MENU_TYPE:{ [index:string] : MenuItem } = {
  RANKING: {
    name: 'Ranking',
    path: '/ranking/daily',
    icon: 'line-chart',
    isActive: true,
  },
  HOT_ENTRY: {
    name: 'HotEntry',
    path: '/hotentry/hotentry',
    icon: 'fire',
    isActive: false,
  },
  ABOUT: {
    name: 'About',
    path: '/about',
    icon: 'info',
    isActive: false,
  }
}
