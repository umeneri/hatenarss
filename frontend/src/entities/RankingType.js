export const RANKING_TYPE = {
  DAILY: {
    period: 'daily',
    name: '日'
  },
  WEEKLY: {
    period: 'weekly',
    name: '週'
  },
  MONTHLY: {
    period: 'monthly',
    name: '月'
  },
  ANNUALY: {
    period: 'annually',
    name: '年'
  },
  ALL: {
    period: 'all',
    name: '累計'
  }
}
