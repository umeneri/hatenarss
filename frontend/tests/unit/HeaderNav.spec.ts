import Vue from 'vue'
import Router from 'vue-router'
import { shallowMount, mount, Wrapper } from '@vue/test-utils';
import HeaderNav from '@/components/molecules/HeaderNav.vue'
import { MENU_TYPE } from '@/entities/MenuType'

Vue.use(Router)

describe('HeaderNav.vue', () => {
  it('toggle menu when button is clicked', () => {
    const wrapper: Wrapper<HeaderNav> = shallowMount(HeaderNav, {
      propsData: { menus: Object.values(MENU_TYPE) }
    })

    expect(wrapper.props().menus[0].name).toBe("Ranking")

    const button = wrapper.find('.navbar-burger');
    expect(button.isVisible).toBeTruthy

    button.trigger('click')

    expect(wrapper.vm.isMenuActive).toBe(true)
    expect(wrapper.html()).toMatchSnapshot();
  })
})
