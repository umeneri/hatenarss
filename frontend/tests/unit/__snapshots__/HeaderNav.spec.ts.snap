// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderNav.vue toggle menu 1`] = `
<header>
  <nav role="navigation" aria-label="main navigation" class="navbar has-shadow">
    <div class="navbar-brand">
      <div class="navbar-item"><img src="/assets/logo.png"></div> <a role="button" aria-label="menu" aria-expanded="false" data-target="navbarManu" class="navbar-burger burger is-active"><span aria-hidden="true"></span> <span aria-hidden="true"></span> <span aria-hidden="true"></span></a>
    </div>
    <div id="navbarManu" class="navbar-menu is-dark is-active">
      <div class="navbar-start">
        <routerlink-stub to="/ranking/daily" tag="a" event="click" class="navbar-item"><span class="icon"><i class="fa fa-line-chart"></i></span> <span>Ranking</span></routerlink-stub>
        <routerlink-stub to="/hotentry/hotentry" tag="a" event="click" class="navbar-item"><span class="icon"><i class="fa fa-fire"></i></span> <span>HotEntry</span></routerlink-stub>
        <routerlink-stub to="/about" tag="a" event="click" class="navbar-item"><span class="icon"><i class="fa fa-info"></i></span> <span>About</span></routerlink-stub>
      </div>
    </div>
  </nav>
</header>
`;

exports[`HeaderNav.vue toggle menu when button is clicked 1`] = `
<header>
  <nav role="navigation" aria-label="main navigation" class="navbar has-shadow">
    <div class="navbar-brand">
      <div class="navbar-item"><img src="/assets/logo.png"></div> <a role="button" aria-label="menu" aria-expanded="false" data-target="navbarManu" class="navbar-burger burger is-active"><span aria-hidden="true"></span> <span aria-hidden="true"></span> <span aria-hidden="true"></span></a>
    </div>
    <div id="navbarManu" class="navbar-menu is-dark is-active">
      <div class="navbar-start">
        <routerlink-stub to="/ranking/daily" tag="a" event="click" class="navbar-item"><span class="icon"><i class="fa fa-line-chart"></i></span> <span>Ranking</span></routerlink-stub>
        <routerlink-stub to="/hotentry/hotentry" tag="a" event="click" class="navbar-item"><span class="icon"><i class="fa fa-fire"></i></span> <span>HotEntry</span></routerlink-stub>
        <routerlink-stub to="/about" tag="a" event="click" class="navbar-item"><span class="icon"><i class="fa fa-info"></i></span> <span>About</span></routerlink-stub>
      </div>
    </div>
  </nav>
</header>
`;

exports[`HeaderNav.vue when toggle menu 1`] = `
<header>
  <nav role="navigation" aria-label="main navigation" class="navbar has-shadow">
    <div class="navbar-brand">
      <div class="navbar-item"><img src="/assets/logo.png"></div> <a role="button" aria-label="menu" aria-expanded="false" data-target="navbarManu" class="navbar-burger burger is-active"><span aria-hidden="true"></span> <span aria-hidden="true"></span> <span aria-hidden="true"></span></a>
    </div>
    <div id="navbarManu" class="navbar-menu is-dark is-active">
      <div class="navbar-start">
        <routerlink-stub to="/ranking/daily" tag="a" event="click" class="navbar-item"><span class="icon"><i class="fa fa-line-chart"></i></span> <span>Ranking</span></routerlink-stub>
        <routerlink-stub to="/hotentry/hotentry" tag="a" event="click" class="navbar-item"><span class="icon"><i class="fa fa-fire"></i></span> <span>HotEntry</span></routerlink-stub>
        <routerlink-stub to="/about" tag="a" event="click" class="navbar-item"><span class="icon"><i class="fa fa-info"></i></span> <span>About</span></routerlink-stub>
      </div>
    </div>
  </nav>
</header>
`;
